### PR TITLE
Remove logback.xml and fix styles

### PR DESCRIPTION
**Pull request by yuanxinwang**
Remove logback.xml and fix styles

## Description
We want users to use their own logback.xml if they need it and they should customize it by themselves.

## Motivation and Context
To configure custom logging behavior. Fix styles.

## How Has This Been Tested?
It passed all existing unit tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Major release (change is NOT backward compatible with prior release)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by yuanxinwang** [comment]
@jui8wang @fansu

**Comment by fansu** [comment]
+1

File: logback.xml [hunk 1]
```diff
@@ -4,14 +4,14 @@
 
     <!-- Primary file appender for server logs -->
     <appender name="R" class="ch.qos.logback.core.rolling.RollingFileAppender">
-        <File>/home/y/logs/yjava_jetty/server.log</File>
+        <File>/tmp/server.log</File>
```
Comment (nituyahoo) [inline logback.xml]: please discuss why tmp?

File: core/src/main/java/com/yahoo/imapnio/async/request/ImapRequest.java [hunk 2]
```diff
@@ -3,8 +3,7 @@
 import java.util.concurrent.ConcurrentLinkedQueue;
 
 import javax.annotation.Nonnull;
-
-import org.apache.avro.reflect.Nullable;
+import javax.annotation.Nullable;
```
Comment (OnePunMan) [inline core/src/main/java/com/yahoo/imapnio/async/request/ImapRequest.java]: Also remove the maven dependency from `pom.xml` since it's no longer needed

Comment (yuanxinwang) [reply core/src/main/java/com/yahoo/imapnio/async/request/ImapRequest.java]: I don't think pom.xml contains avro since it is common used library.

**Pull request merged**